src/main/java/org/reactome/server/controller/DoiController.java

```diff
@@ -0,0 +1,83 @@
+package org.reactome.server.controller;
+
+
+import org.reactome.server.graph.exception.CustomQueryException;
+import org.reactome.server.graph.service.*;
+import org.reactome.server.result.CustomPathway;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import java.util.*;
+
+
+@SuppressWarnings("unused")
+@Controller
+class DoiController {
+
+    private static final Logger errorLogger = LoggerFactory.getLogger("errorLogger");
+
+    private static final String TITLE = "title";
+    private static final String DOIPATHWAYS = "doiPathways";
+
+    private AdvancedDatabaseObjectService advancedDatabaseObjectService;
+
+
+    @Autowired
+    public void setAdvancedDatabaseObjectService(AdvancedDatabaseObjectService advancedDatabaseObjectService) {
+        this.advancedDatabaseObjectService = advancedDatabaseObjectService;
+    }
+
+    public DoiController() {
+
+    }
+
+    @RequestMapping(value = "/doi", method = RequestMethod.GET)
+    public String doiTable(ModelMap model) {
+
+        Collection<CustomPathway> doiPathways = getAllDoiPathways();
+
+        model.addAttribute(TITLE, "DOI Table");
+        model.addAttribute(DOIPATHWAYS, doiPathways);
+        return "graph/doi";
+    }
+
+    /**
+     * Retrieve all pathway which the doi value is not null, and all the authors,reviewers, editors are
+     * related  to this pathway, subpathway and reactions.
+     */
+    public Collection<CustomPathway> getAllDoiPathways() {
+        Collection<CustomPathway> rtn;
+
+        String query = "MATCH (p:Pathway)" +
```

Comment:
hmm, are you going to keep the query in the controller ?

Reply:
no, just for now